File: my_cpp_service/src/pow_two_ints_client.cpp
#include "rclcpp/rclcpp.hpp"
#include "my_interfaces/srv/pow_two_ints.hpp"

#include <chrono>
#include <cstdlib>
#include <memory>

using namespace std::chrono_literals;

int main(int argc, char **argv)
{
  rclcpp::init(argc, argv);

  if (argc != 3) {
      RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "usage: pow_two_ints_client X Y");
      return 1;
  }

  std::shared_ptr<rclcpp::Node> node = rclcpp::Node::make_shared("pow_two_ints_client");
  rclcpp::Client<my_interfaces::srv::PowTwoInts>::SharedPtr client =
    node->create_client<my_interfaces::srv::PowTwoInts>("pow_two_ints");

  auto request = std::make_shared<my_interfaces::srv::PowTwoInts::Request>();
  request->a = atoll(argv[1]);
  request->b = atoll(argv[2]);

  while (!client->wait_for_service(1s)) {
    if (!rclcpp::ok()) {
      RCLCPP_ERROR(rclcpp::get_logger("rclcpp"), "Interrupted while waiting for the service. Exiting.");
      return 0;
    }
    RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "service not available, waiting again...");
  }

  auto result = client->async_send_request(request);
  // Wait for the result.
  if (rclcpp::spin_until_future_complete(node, result) ==
    rclcpp::FutureReturnCode::SUCCESS)
  {
    RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "c: %ld", result.get()->c);
  } else {
    RCLCPP_ERROR(rclcpp::get_logger("rclcpp"), "Failed to call service pow_two_ints");
  }

  rclcpp::shutdown();
  return 0;
}

File: my_py_service/my_py_service/pow_two_ints_client.py
import sys

from my_interfaces.srv import PowTwoInts
import rclpy
from rclpy.node import Node

class PowTwoIntsClientAsync(Node):

    def __init__(self):
        super().__init__("pow_two_ints_client")
        self.cli = self.create_client(PowTwoInts, "pow_two_ints")
        while not self.cli.wait_for_service(timeout_sec=1.0):
            self.get_logger().info('service not available, waiting again...')
        self.req = PowTwoInts.Request()

    def send_request(self, a, b):
        self.req.a = a
        self.req.b = b
        self.future = self.cli.call_async(self.req)
        rclpy.spin_until_future_complete(self, self.future)
        return self.future.result()


def main(args=None):
    rclpy.init(args=args)

    pow_two_ints_client = PowTwoIntsClientAsync()
    response = pow_two_ints_client.send_request(int(sys.argv[1]), int(sys.argv[2]))
    pow_two_ints_client.get_logger().info(
        'Result of pow_two_ints: for 2^%d + 3^%d = %d' %
        (int(sys.argv[1]), int(sys.argv[2]), response.c))

    pow_two_ints_client.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()
